aspnetcore/fundamentals/configuration/index.md

```diff
@@ -1,452 +1,1308 @@
+
+* section0:key0
+* section0:key1
+* section1:key0
+* section1:key1
+
+[GetSection](/dotnet/api/microsoft.extensions.configuration.iconfiguration.getsection) and [GetChildren](/dotnet/api/microsoft.extensions.configuration.iconfiguration.getchildren) methods are available to isolate sections and children of a section in the configuration data. These methods are described later in [GetSection, GetChildren, and Exists](#getsection-getchildren-and-exists).
+
+## Conventions
+
+At app startup, configuration sources are read in the order that their configuration providers are specified. The last provider to set a value for a key determines the value stored in configuration at startup. 
+
+File Configuration Providers have the ability to reload configuration when an underlying settings file is changed after startup. The File Configuration Provider is described later in this topic.
+
+[Dependency Injection (DI)](xref:fundamentals/dependency-injection) isn't functional until `Startup.ConfigureServices` executes. Therefore, the configuration system can't interact with DI until `ConfigureServices` executes.
+
+Configuration keys adopt the following conventions:
+
+* Keys are case-insensitive. For example, `ConnectionString` and `connectionstring` are treated as equivalent keys.
+* If a value for the same key is set by the same or different configuration providers, the last value set on the key is the value used.
+* Hierarchical keys
+  * Within the Configuration API, a colon separator (`:`) works on all platforms.
+  * In environment variables, a colon separator (`:`) may not work on all platforms. A double underscore (`__`) is supported by all platforms.
+  * In Azure Key Vault, hierarchical keys use `--` (two dashes) as a separator.
 
-These packages are included in the [Microsoft.AspNetCore.All metapackage](xref:fundamentals/metapackage).
+Configuration values are strings.
+
+## Security
+
+Never store passwords or other sensitive data in configuration provider code or in plain text configuration files. Don't use production secrets in development or test environments. Specify secrets outside of the project so that they can't be accidentally committed to a source code repository.
+
+Learn more about [how to use multiple environments](xref:fundamentals/environments) and managing the [safe storage of app secrets in development with the Secret Manager](xref:security/app-secrets) (includes advice on using environment variables to store sensitive data). The Secret Manager uses the File Configuration Provider to store user secrets in a JSON file on the local system. The File Configuration Provider is described later in this topic.
+
+[Azure Key Vault](https://azure.microsoft.com/services/key-vault/) is one option for the safe storage of app secrets. For more information, see <xref:security/key-vault-configuration>.
+
+## Providers
+
+The configuration providers available from the framework and described in this topic are shown in the following table.
```

Comment:
I'm ok with that approach, but the reader should receive some kind of tipoff that the content isn't in this topic. Otherwise, they may assume that they'll get full coverage here. I'll try a format that includes "(external)," and we can see how that looks on the next commit. I'll add User Secrets to the list and link out ... that was just an oversight ... I have it elsewhere here.